lib/ecto/query/planner.ex

```diff
@@ -2283,11 +2290,14 @@ defmodule Ecto.Query.Planner do
       {source, schema, _} = all when is_binary(source) and schema != nil ->
         all
 
+      %Ecto.SubQuery{} = subquery when ix != 0 ->
+        subquery
+
       _ ->
         error!(
           query,
-          "can only preload sources with a schema " <>
-            "(fragments, binary and subqueries are not supported)"
+          "can only preload sources with a schema or subqueries in `join`" <>
```

Comment:
If we are assuming it defaults to the association, then we can allow anything, including fragments, no?

Reply:
Definitely, will fully review when back at home